jennchenn/uwexplore
# Fix CSE calculation, color assignment, search hierarchy

## Implementation description
random backend bug fixes from user testing
- calculate CSE CEAB totals when adding a course (previously always 0)
- show department first if search keyword matches a department
- assign colors randomly to a course in the backend
- resolves BREAD-159



## Steps to test
_CEAB_
1. Add CSEs to the schedule/past courses
2. Verify that the CSE totals and the TE+CSE totals update to reflect changes

_Search hierarchy_
1. Search for a course by department
2. Verify courses that match that department name appear first (e.g. `PD`/`SYDE`)
3. Sanity check that search still works as normal

_Color assignment_
1. Add a bunch of courses to schedule
2. Verify tut/lec have same color, colors are assigned uniquely



## What should reviewers focus on?

- when all colors have been used, the color assignment loops from the beginning - there are some issues w this bc it doesn't account for courses that don't appear on the schedule (i.e. online courses) -> hopefully it is still unique enough that it's okay?
